### PR TITLE
Expanding of env variables in input_file_mapping and others

### PR DESCRIPTION
**Pull request by pwl**

### CODE REVIEW

**Comment by pwl** [comment]
I'm expanding the environmental variables in `FG_DATA_ROOT`, `FG_APP_DIR` & for each item in the `INPUT_FILE_MAPPING`.  There is a new test directory for that, `test.env`, the tests had to be separated as settingn the environmental variables should override files/directories passed through the arguments to `FGApp`, `FGData` & `FGProcess`.  To run the tests use `bash tests.env/run.sh`, the new tests are integrated into the travis.

File: fastgenomics/io.py [hunk 1]
```diff
@@ -37,7 +38,7 @@ def __init__(self, specs, root, mapping):
             path = None
             isoptional = "optional" in spec and spec["optional"]
             if name in mapping:
-                path = Path(root) / mapping[name]
+                path = Path(os.path.expandvars(Path(root) / mapping[name]))
```
Comment (schlemmge) [inline fastgenomics/io.py]: Are the brackets placed properly? Do we really need to expand mapping[name] as well?

Comment (pwl) [reply fastgenomics/io.py]: good find, this was the initial version I implemented but then I decided to move the path expansions out of this function into the defaulst.py, this `os.path.expandvars` should not be there.

Comment (pwl) [reply fastgenomics/io.py]: I restored it back to the original.

**Pull request merged**